rust/arrow/src/ipc/writer.rs

```diff
@@ -434,6 +434,12 @@ impl<W: Write> FileWriter<W> {
 
     /// Write footer and closing tag, then mark the writer as done
     pub fn finish(&mut self) -> Result<()> {
+        if self.finished {
```

Comment:
This is just a drive-by fix to defend against double-writes of the footer I noticed while making the change.

Reply:
this  looks like a good change to to me ( protect people from themselves)